test/unittests/state_tx_test.cpp

```diff
@@ -153,3 +154,39 @@ TEST(state_tx, validate_blob_tx)
                   .message(),
         make_error_code(ErrorCode::INVALID_BLOB_HASH_VERSION).message());
 }
+
+TEST(state_tx, validate_data)
+{
+    const BlockInfo bi{.gas_limit = 0x989680,
+        .coinbase = 0x01_address,
+        .prev_randao = {},
+        .base_fee = 0x0a,
+        .withdrawals = {}};
+    const Account acc{.nonce = 1, .balance = 0xe8d4a51000};
+    Transaction tx{
+        .data = "EF00 01 010004 0200010001 030004 00 00000000 00 AABBCCDD"_hex,
+        .gas_limit = 60000,
+        .max_gas_price = bi.base_fee,
+        .max_priority_gas_price = 0,
+        .sender = 0x02_address,
+        .to = {},
+        .value = 0,
+        .access_list = {},
+        .nonce = 1,
+        .r = 0,
+        .s = 0,
+    };
+
+    ASSERT_FALSE(holds_alternative<std::error_code>(
+        validate_transaction(acc, bi, tx, EVMC_CANCUN, 60000, BlockInfo::MAX_BLOB_GAS_PER_BLOCK)));
+
+    EXPECT_EQ(std::get<std::error_code>(validate_transaction(acc, bi, tx, EVMC_PRAGUE, 60000,
+                                            BlockInfo::MAX_BLOB_GAS_PER_BLOCK))
+                  .message(),
+        "EOF in creation transaction");
```

Comment:
```suggestion
        "EOF initcode in creation transaction");
```

to be more specific? Can also be "in creation transaction data"